lib/connection.js

```diff
@@ -626,7 +626,15 @@ Connection.prototype.close = function (callback) {
   this.isSocketOpen = false;
   this.log('verbose', 'Closing connection to ' + this.endpoint);
   const self = this;
+
+  // If server doesn't acknowledge the half-close within connection timeout, destroy the socket.
+  const endTimeout = setTimeout(() => {
+    self.log('info', this.endpoint + ' did not respond to connection close within ' + this.options.socketOptions.connectTimeout + 'ms, destroying connection');
```

Comment:
Minor: mix use of `self` and `this`.

Reply:
D'oh!  That's a bad habit of mine, will fix :)